doc/sphinxext/gallery_generator.py

```diff
@@ -269,6 +287,7 @@ def contents_entry(self) -> str:
             pngfilename=self.pngfilename,
             sphinx_tag=self.sphinxtag,
             title=self.title,
+            overlay="{0}{1}".format(self.title, (" using " + self.apiname) if self.apiname != "No API Documentation available" else ""),
```

Comment:
Same thing with overlay. Create a dedicated method in the class.

You are a one liner pro though, I am so impressed by your ability to cram so much logic in 80 characters